core/devdoc/gateway_createfromjson_requirements.md

```diff
@@ -118,3 +118,30 @@ Gateway_CreateFromJson creates a new gateway using information contained within
 **SRS_GATEWAY_JSON_17_002: [** This function shall return `NULL` if starting the gateway fails. **]**
 
 **SRS_GATEWAY_JSON_14_008: [** This function shall return `NULL` upon any memory allocation failure. **]**
+
+
+## Gateway_UpdateFromJson
+```
+extern int Gateway_UpdateFromJson(GATEWAY_HANDLE gw, const char* json_content);
+```
+Gateway_UpdateFromJson adds loades, modules or links based on a well-formed JSON configuration content.
+
+**SRS_GATEWAY_JSON_04_003: [** If `json_content` is NULL the function shall return NULL. **]**
+
+**SRS_GATEWAY_JSON_04_004: [** If `gw` is NULL the function shall return NULL.
+ **]**
+
+**SRS_GATEWAY_JSON_04_005: [** The function shall use *parson* to parse the JSON string to a *parson* `JSON_Value` structure. **]**
+
+**SRS_GATEWAY_JSON_04_006: [** The function shall return NULL if the JSON content could not be parsed to a `JSON_Value`. **]**
+
+**SRS_GATEWAY_JSON_04_007: [** The function shall traverse the `JSON_Value` object to initialize a `GATEWAY_PROPERTIES` instance. **]**
+
+**SRS_GATEWAY_JSON_04_010: [** The function shall return NULL if the `JSON_Value` contains incomplete information. **]**
+
+**SRS_GATEWAY_JSON_04_011: [** The function shall be able to add just `modules` or just `links`.
```

Comment:
Nops, that didn't change, It can be just module, or just links (if module exists) or both.... 

---
In reply to: [108960341](https://github.com/Azure/azure-iot-gateway-sdk/pull/189#discussion_r108960341) [](ancestors = 108960341)